content/webapp/utils/urls.ts

```diff
@@ -1,3 +1,4 @@
+// TODO modify logic for this. Should we make UID the canonical link?
```

Comment:
To discuss with team tomorrow

Reply:
@wellcomecollection/digital-experience I've addressed this [here](https://github.com/wellcomecollection/wellcomecollection.org/pull/11288/commits/02dff514426ac70a813aa33d38eadea1eb7df88c).

This helper function was used in setting the Canonical URLs of some pages, returning `true` if the URL was not a Prismic ID and matched "a single alphanumeric slug directly after the /", which our UID-based URLs do. That meant that if I was accessing a page with various valid UIDs for it, it would change what was set as its canonical URL every time.

So wherever it was used, I changed it to set it to the document's `uid`, which is always the value of its current one (whereas the URL could be a former valid `uid` of that document).

I then have gone into the few pages that I think _could_ use vanity URLs, which is the `pages` in our route whose URL we've determined by how the file is name, who then fetch a document from Prismic based on the ID we give them. In those cases, I've made sure the canonical link was set to whatever the file was named.

I hope this makes sense, will await for your feedback before merging.